### PR TITLE
Mobile 3833

### PR DESCRIPTION
**Pull request by crazyserver**

### CODE REVIEW

**Review by dpalou — CHANGES_REQUESTED** [review]

File: src/core/features/course/components/section-selector/section-selector.ts [hunk 1]
```diff
@@ -56,14 +56,13 @@ export class CoreCourseSectionSelectorComponent implements OnInit {
             let complete = 0;
             let total = 0;
             section.modules.forEach((module) => {
-                if (!module.uservisible || module.completiondata === undefined || module.completiondata.tracking === undefined ||
-                        module.completiondata.tracking <= CoreCourseProvider.COMPLETION_TRACKING_NONE) {
+                if (!module.uservisible || module.completiondata === undefined || module.completiondata.tracking === undefined) {
```
[the remark this answers is not in the document]
Comment (crazyserver) [reply src/core/features/course/components/section-selector/section-selector.ts]: Nice catch!

File: src/core/features/course/services/course-helper.ts [hunk 2]
```diff
@@ -2093,7 +2097,6 @@ export class CoreCourseHelperProvider {
 
             return response;
         } catch (error) {
-            completion.state = completion.state === 1 ? 0 : 1;
```
[the remark this answers is not in the document]
Comment (crazyserver) [reply src/core/features/course/services/course-helper.ts:2096]: Oh yeah! I didn't get the toggle in the code... thanks!

File: src/core/features/course/services/course.ts [hunk 3]
```diff
@@ -1474,7 +1482,12 @@ export type CoreCourseGetContentsParams = {
 /**
  * Data returned by core_course_get_contents WS.
  */
-export type CoreCourseWSSection = {
+type CoreCourseGetContentsWS = CoreCourseGetContentsWSSection[];
```
[the remark this answers is not in the document]
Comment (crazyserver) [reply src/core/features/course/services/course.ts:1510]: Totally true!

File: src/core/features/course/services/course-helper.ts [hunk 4]
```diff
@@ -1429,26 +1429,6 @@ export class CoreCourseHelperProvider {
...
 
-    /**
-     * Get the course ID from a module instance ID, showing an error message if it can't be retrieved.
-     *
-     * @param id Instance ID.
-     * @param module Name of the module. E.g. 'glossary'.
-     * @param siteId Site ID. If not defined, current site.
-     * @return Promise resolved with the module's course ID.
-     */
-    async getModuleCourseIdByInstance(id: number, module: string, siteId?: string): Promise<number> {
```
[the remark this answers is not in the document]
Comment (crazyserver) [reply src/core/features/course/services/course-helper.ts:1443]: Yep thx!

File: src/addons/mod/h5pactivity/services/handlers/report-link.ts [hunk 5]
```diff
@@ -41,27 +41,24 @@ export class AddonModH5PActivityReportLinkHandlerService extends CoreContentLink
...
         url: string,
         params: Record<string, string>,
-        courseId?: number,
     ): CoreContentLinksAction[] | Promise<CoreContentLinksAction[]> {
-        courseId = courseId || Number(params.courseid) || Number(params.cid);
-
         return [{
             action: async (siteId) => {
                 try {
                     const instanceId = Number(params.a);
```
Comment (dpalou) [inline src/addons/mod/h5pactivity/services/handlers/report-link.ts:49]: Shouldn't you display a loading modal here? Before, it was done inside getCourseId.

**Review by dpalou — CHANGES_REQUESTED** [review]

File: src/addons/mod/h5pactivity/services/handlers/report-link.ts (hunk 5 above)
Comment (crazyserver) [reply src/addons/mod/h5pactivity/services/handlers/report-link.ts:51]: Done!

**Review by dpalou — APPROVED** [review]

**Pull request merged**